staging/src/k8s.io/kube-apiextensions-server/pkg/apiserver/customresource_handler.go

```diff
@@ -0,0 +1,366 @@
+
+	requestScope := handlers.RequestScope{
+		Namer: handlers.ContextBasedNaming{
+			GetContext: func(req *http.Request) apirequest.Context {
+				ret, _ := r.requestContextMapper.Get(req)
+				return ret
+			},
+			SelfLinker:    meta.NewAccessor(),
+			ClusterScoped: customResource.Spec.Scope == apiextensions.ClusterScoped,
+		},
+		ContextFunc: func(req *http.Request) apirequest.Context {
+			ret, _ := r.requestContextMapper.Get(req)
+			return ret
+		},
+
+		Serializer:     UnstructuredNegotiatedSerializer{},
+		ParameterCodec: parameterCodec,
+
+		Creater:         UnstructuredCreator{},
+		Convertor:       unstructured.UnstructuredObjectConverter{},
+		Defaulter:       UnstructuredDefaulter{},
+		Copier:          UnstructuredCopier{},
+		Typer:           discovery.NewUnstructuredObjectTyper(nil),
+		UnsafeConvertor: unstructured.UnstructuredObjectConverter{},
+
+		Resource:    schema.GroupVersionResource{Group: customResource.Spec.Group, Version: customResource.Spec.Version, Resource: customResource.Spec.Names.Plural},
+		Kind:        schema.GroupVersionKind{Group: customResource.Spec.Group, Version: customResource.Spec.Version, Kind: customResource.Spec.Names.Kind},
+		Subresource: "",
+
+		MetaGroupVersion: metav1.SchemeGroupVersion,
+	}
+
+	ret = &customResourceInfo{
+		storage:      storage,
+		requestScope: requestScope,
+	}
+	storageMap[customResource.UID] = ret
+	r.customStorage.Store(storageMap)
+	return ret
+}
+
+type UnstructuredNegotiatedSerializer struct{}
+
+func (s UnstructuredNegotiatedSerializer) SupportedMediaTypes() []runtime.SerializerInfo {
+	return []runtime.SerializerInfo{
+		{
+			MediaType:        "application/json",
+			EncodesAsText:    true,
+			Serializer:       json.NewSerializer(json.DefaultMetaFactory, UnstructuredCreator{}, discovery.NewUnstructuredObjectTyper(nil), false),
+			PrettySerializer: json.NewSerializer(json.DefaultMetaFactory, UnstructuredCreator{}, discovery.NewUnstructuredObjectTyper(nil), true),
+			StreamSerializer: &runtime.StreamSerializerInfo{
+				EncodesAsText: true,
+				Serializer:    json.NewSerializer(json.DefaultMetaFactory, UnstructuredCreator{}, discovery.NewUnstructuredObjectTyper(nil), false),
+				Framer:        json.Framer,
+			},
+		},
+	}
+}
+
+func (s UnstructuredNegotiatedSerializer) EncoderForVersion(serializer runtime.Encoder, gv runtime.GroupVersioner) runtime.Encoder {
+	return unstructured.UnstructuredJSONScheme
+}
+
+func (s UnstructuredNegotiatedSerializer) DecoderToVersion(serializer runtime.Decoder, gv runtime.GroupVersioner) runtime.Decoder {
+	return unstructured.UnstructuredJSONScheme
+}
+
+type UnstructuredCreator struct{}
+
+func (UnstructuredCreator) New(kind schema.GroupVersionKind) (runtime.Object, error) {
+	ret := &unstructured.Unstructured{}
+	ret.SetGroupVersionKind(kind)
+	return ret, nil
+}
+
+type UnstructuredCopier struct{}
+
+func (UnstructuredCopier) Copy(obj runtime.Object) (runtime.Object, error) {
```

Comment:
you can steal a better deepcopy from https://github.com/kubernetes/kubernetes/pull/40725